test/api.test.js

```diff
@@ -198,12 +198,18 @@ describe('API', () => {
   });
 
   describe('context property', () => {
-    it('should contain public properties', () => {
+    it('should contain public properties', (done) => {
       expect(instance.context.state).toBeDefined();
       expect(instance.context.options).toBeDefined();
       expect(instance.context.compiler).toBeDefined();
       expect(instance.context.watching).toBeDefined();
       expect(instance.context.outputFileSystem).toBeDefined();
+
+      // the compilation needs to finish, as it will still be running
+      // after the test is done if not finished, potentially impacting other tests
+      compiler.hooks.done.tap('wdm-test', () => {
+        done();
```

Comment:
I couldn't reproduce the problem locally, but there was a problem with `api.test.js` impacting `getFilenameFromUrl.test.js` when it was being run directly after `api.test.js`, happening on CI for webpack@next only.

The problem was that the very last test of `api.test.js` was not waiting for compilation to finish, and somehow this was impacting the very first test of `getFilenameFromUrl.test.js` by preventing the middleware's done hook from being called for this very first test.

I'm not sure why this was happening, but it could be something with webpack@next saving some global data about compiler hooks that is still there when the new webpack compilation starts. Also still not sure why I couldn't reproduce locally, I tried to make everything identical to the CI environment and installed webpack@next but tests still passed.

Reply:
Very strange, if I comment this line, it is happens in CI, right?

Reply:
@evilebottnawi Yes